### PR TITLE
Update README to use the new config syntax

### PR DESCRIPTION
**Pull request by carlos-granados**
Use `RectorConfig::configure()` instead of `static function (RectorConfig $rectorConfig)`

There are related PRs in the 

- Rector-symfony respository: https://github.com/rectorphp/rector-symfony/pull/585
- Rector-phpunit repository: https://github.com/rectorphp/rector-phpunit/pull/314
- Rector-doctrine repository: https://github.com/rectorphp/rector-doctrine/pull/293

### CODE REVIEW

File: build/target-repository/README.md [hunk 1]
```diff
@@ -46,16 +46,16 @@ use Rector\Config\RectorConfig;
...
     // register single rule
-    $rectorConfig->rule(TypedPropertyFromStrictConstructorRector::class);
-
+    ->withRules([
+        TypedPropertyFromStrictConstructorRector::class
+    ])
     // here we can define, what sets of rules will be applied
     // tip: use "SetList" class to autocomplete sets with your IDE
-    $rectorConfig->sets([
+    ->withSets([
```
Comment (samsonasik) [inline build/target-repository/README.md]: there is `withPreparedSets` for codeQuality, ref

https://github.com/rectorphp/rector-src/blob/827992ae9784abb4a2a5f5f3aecbe52606f3c737/rector.php#L14-L16

Comment (carlos-granados) [reply build/target-repository/README.md]: updated

Comment (samsonasik) [reply build/target-repository/README.md]: Hmm, github diff seems not updated, probably a github glicth lately? as I got repeated same notification

Comment (carlos-granados) [reply build/target-repository/README.md]: Yeah, there is something weird going on with Github. Previously it was not seeing this branch. Let me try again

Comment (samsonasik) [reply build/target-repository/README.md]: no worries, github status shown it down now https://www.githubstatus.com/

Comment (samsonasik) [reply build/target-repository/README.md]: for `withPreparedSet`, you need type:

```php
withPreparedSet(codeQuality: true)
```

You can add another setlist, eg: from phpunit with `withSets()`

Comment (carlos-granados) [reply build/target-repository/README.md]: Is it OK now?

Comment (carlos-granados) [reply build/target-repository/README.md]: Ah, no, sorry, I'll update

Comment (carlos-granados) [reply build/target-repository/README.md]: hopefully now it should be OK, let me know

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
@TomasVotruba let's merge it, thank you @carlos-granados
